sdk/identity/azure-identity/src/test/java/com/azure/identity/EnvironmentCredentialTests.java

```diff
@@ -36,4 +36,46 @@ public void testCreateEnvironmentCredential() {
             .expectNextMatches(token -> "token".equals(token.getToken()))
             .verifyComplete();
     }
+
+    @Test
+    public void testCreateEnvironmentClientCertificateCredential() {
+        Configuration configuration = Configuration.getGlobalConfiguration();
+        configuration.put(Configuration.PROPERTY_AZURE_CLIENT_ID, "foo");
+        configuration.put(Configuration.PROPERTY_AZURE_CLIENT_CERTIFICATE_PATH, "bar");
+        configuration.put(Configuration.PROPERTY_AZURE_TENANT_ID, "baz");
+
+        EnvironmentCredential credential = new EnvironmentCredentialBuilder().build();
+
+        // authentication will fail client-id=foo, but should be able to create ClientSecretCredential
```

Comment:
NIT: I think this comment should say "ClientCertificateCredential" instead of "ClientSecretCredential"